x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Block.java

```diff
@@ -212,10 +212,44 @@ default boolean mvSortedAscending() {
     /**
      * Expand multivalued fields into one row per value. Returns the same block if there aren't any multivalued
      * fields to expand. The returned block needs to be closed by the caller to release the block's resources.
-     * TODO: pass BlockFactory
      */
     Block expand();
 
+    /**
+     * Build a {@link Block} with a {@code null} inserted {@code before} each
+     * listed position.
+     * <p>
+     *     Note: {@code before} must be non-decreasing.
+     * </p>
+     */
+    default Block insertNulls(IntVector before) {
+        // TODO remove default and scatter to implementation where it can be a lot more efficient
+        try (Builder builder = elementType().newBlockBuilder(getPositionCount() + before.getPositionCount(), blockFactory())) {
```

Comment:
Nit: 
Since there are multiple calls to getPositionCount() and before.getPositionCount() how about extracting them into separate vars:
```
try (long pc = getPositionCount(), beforePC = before.getPositionCount(), Builder builder = ...) { } 
```

Reply:
Done.